rest/test_results_routes.go

```diff
@@ -14,8 +26,56 @@ package rest
 //
 // GET /testresults/test_name/{task_id}/{test_name}
 
-// Factory
+type testResultGetByTestNameHandler struct {
+	opts data.TestResultsOptions
+	sc   data.Connector
+}
 
-// Parse
+func makeGetTestResultByTestName(sc data.Connector) gimlet.RouteHandler {
+	return &testResultGetByTestNameHandler{
+		sc: sc,
+	}
+}
 
-// Run
+// Factory returns a pointer to a new testResultGetByTestNameHandler.
+func (h *testResultGetByTestNameHandler) Factory() gimlet.RouteHandler {
+	return &testResultGetByTestNameHandler{
+		sc: h.sc,
+	}
+}
+
+// Parse fetches the task_id, test_name, and execution (if present)
+// from the http request.
+func (h *testResultGetByTestNameHandler) Parse(_ context.Context, r *http.Request) error {
+	catcher := grip.NewBasicCatcher()
+	var err error
+
+	h.opts.TaskID = gimlet.GetVars(r)["task_id"]
+	h.opts.TestName = gimlet.GetVars(r)["test_name"]
+	vals := r.URL.Query()
+	if len(vals[execution]) > 0 {
+		h.opts.Execution, err = strconv.Atoi(vals[execution][0])
+		catcher.Add(err)
+	} else {
+		h.opts.EmptyExecution = true
+	}
+
+	return catcher.Resolve()
+}
+
+// Run finds and returns the desired test result.
+func (h *testResultGetByTestNameHandler) Run(ctx context.Context) gimlet.Responder {
+	testResult, err := h.sc.FindTestResultByTestName(ctx, h.opts)
+	if err != nil {
+		err = errors.Wrapf(err, "problem getting test result by task_id '%s' and test_name '%s'", h.opts.TaskID, h.opts.TestName)
+		grip.Error(message.WrapError(err, message.Fields{
+			"request":   gimlet.GetRequestID(ctx),
+			"method":    "GET",
+			"route":     "/testresults/test_name/{task_id}/{test_name}",
+			"task_id":   h.opts.TaskID,
+			"test_name": h.opts.TestName,
+		}))
+		return gimlet.MakeJSONErrorResponder(err)
```

Comment:
I realize you probably copied this, but if you mean to return internal server error from the request, you have to use `gimlet.MakeJSONInternalErrorResponder()` (`gimlet.MakeJSONErrorResponder()` returns bad request). 

@julianedwards do you actually want to return bad request and not internal server error?

Reply:
No this should be internal server error, good catch